numpy/lib/function_base.py

```diff
@@ -3590,11 +3610,19 @@ def kaiser(M, beta):
     >>> plt.show()
 
     """
+    # Ensures at least float64 via 0.0.  M should be an integer, but conversion
+    # to double is safe for a range.  (Simplified result_type with 0.0
+    # strongly typed.  result-type is not/less order sensitive, but that mainly
+    # matters for integers anyway.)
+    values = np.array([0.0, M, beta])
```

Comment:
result-type can actually deal with the fact that `uint16, int16, float32` should promote to float32, but `np.array([uint16(1), int16(1), float32(1)])` does not (it would return float64 there because `uint16` and `int16` promote to `int32` first).

I am ignoring that, its mostly relevant for ints, which is irrelevant here.  For user dtypes it could be in theory, but its niche enough that I will consider it a future bug fix if anyone ever notices before the dawn of time.